nixos/modules/services/hardware/pcscd.nix

```diff
@@ -68,18 +77,19 @@ in
     };
   };
 
-  config = lib.mkIf config.services.pcscd.enable {
+  config = lib.mkIf cfg.enable {
     environment.etc."reader.conf".source = cfgFile;
 
-    environment.systemPackages = [ package ];
-    systemd.packages = [ package ];
+    environment.systemPackages = [ cfg.package ];
+    systemd.packages = [ cfg.package ];
 
     services.pcscd.plugins = [ pkgs.ccid ];
 
     systemd.sockets.pcscd.wantedBy = [ "sockets.target" ];
 
     systemd.services.pcscd = {
       environment.PCSCLITE_HP_DROPDIR = pluginEnv;
+      reloadTriggers = [ cfgFile ];
```

Comment:
This is not working correctly because `cfgFile` is used in `ExecStart`. The unit will automatically restart instead of reloading whenever `cfgFile` changes its hash